src/queryMetrics/clientSideMetrics.ts

```diff
@@ -0,0 +1,58 @@
+import { QueryMetricsUtils } from "./queryMetricsUtils";
+
+export class ClientSideMetrics {
+    private requestCharge: number;
+    constructor(requestCharge: number) {
+        // Constructor
+        if (!QueryMetricsUtils.isNumeric(requestCharge)) {
+            throw new Error("requestCharge is not a numeric type");
+        }
+
+        this.requestCharge = requestCharge;
+    }
+
+    /**
+     * Gets the RequestCharge
+     * @memberof ClientSideMetrics
+     * @instance
+     * @ignore
+     */
+    public getRequestCharge() {
+        return this.requestCharge;
+    }
+
+    /**
+     * Adds one or more ClientSideMetrics to a copy of this instance and returns the result.
+     * @memberof ClientSideMetrics
+     * @instance
+     */
+    public add(clientSideMetricsArray: ClientSideMetrics[]) {
```

Comment:
Can we use this:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters

since the original intent was for a variable length number of args.